docs-md/concepts/processing-guarantees.md

```diff
@@ -1,11 +1,93 @@
 ---
 layout: page
 title: Processing Guarantees
-tagline: Processing in ksqlDB
-description: Learn about processing in ksqlDB
-keywords: ksqldb, processing
+tagline: Processing guarantees in ksqlDB
+description: Learn about at-least-once, exactly-once semantics in ksqlDB
+keywords: ksqldb, processing, eos, at-least-once, exactly-once
 ---
 
-TODO: EOS, etc.
+ksqlDB supports *at-least-once* and *exactly-once* processing guarantees.
 
-Page last revised on: {{ git_revision_date }}
\ No newline at end of file
+At-least-once semantics
+-----------------------
+
+Records are never lost but may be redelivered. If your stream processing
+application fails, no data records are lost and fail to be processed, but some
+data records may be re-read and therefore re-processed. At-least-once semantics
+is enabled by default in your ksqlDB configuration, with
+`processing.guarantee="at_least_once"`.
+
+Exactly-once semantics
+----------------------
+
+Records are processed once. Even if a producer sends a duplicate record, it's

```

Comment:
I realize this is just cut-and-paste but I found this confusing the first time I read it: "producer" here doesn't refer to the application producer (the initial producer before the stream processing takes place, the analogous producer to the "end consumer"), but rather only producers within KSQL / Streams apps. 

As a more general comment along the same lines, I wonder if we should clarify somewhere that exactly-once only refers to the stream processing guarantees and needs to be used with care (end consumers and producers that are also exactly-once) in order to achieve a truly exactly-once system?

Reply:
Cool, I added a warning note.